bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/WebClientFactoryImpl.java

```diff
@@ -96,30 +100,45 @@ protected void deactivate() {
             try {
                 commonHttpClient.stop();
             } catch (Exception e) {
-                logger.error("error while stopping shared jetty http client", e);
+                logger.error("error while stopping shared Jetty http client", e);
                 // nothing else we can do here
             }
             commonHttpClient = null;
             threadPool = null;
-            logger.debug("jetty shared http client stopped");
+            logger.debug("Jetty shared http client stopped");
         }
     }
 
     @Override
     public HttpClient createHttpClient(@NonNull String consumerName, @NonNull String endpoint) {
         Objects.requireNonNull(endpoint, "endpoint must not be null");
-        logger.debug("httpClient for endpoint {} requested", endpoint);
+        logger.debug("http client for endpoint {} requested", endpoint);
         checkConsumerName(consumerName);
         return createHttpClientInternal(consumerName, endpoint, false);
     }
 
+    @Override
+    public @NonNull WebSocketClient createWebSocketClient(@NonNull String consumerName, @NonNull String endpoint) {
```

Comment:
I just copied it over from the existing methods, so that it looked the same as the rest.

I now tried to add `@NonNullByDefault` to the class, but I failed to get the code to compile due to the fact that this annotation falsely declares the `PrivilegedExceptionAction.run` method to return a non-null value. I have thus reverted all my changes again and kept the annotations as they are. I guess that's also the reason why the code already is as it is.
If you have a good solution that does not do false declarations, feel free to do a follow-up PR with it.

Reply:
Created a PR for your branch.